python/deepclient/deep_client.py

```diff
@@ -448,17 +451,197 @@ def default_returning(self, table: str) -> str:
+        del m[table]
+        return {**m, "data": data['returning']}
 
-    async def serial(self):
-        raise NotImplementedError("Method not implemented")
+    async def update(self, condition: Dict, record: Dict, options: Dict = {}) -> Dict:
+        if not condition or not record:
+            return {"error": {"message": "!condition or !record"}, "data": None, "loading": False,
+                    "networkStatus": None}
+
+        table = options.get("table", self.table)
+        returning = options.get("returning", self.default_returning(table))
+        name = options.get("name", "UPDATE")
+
+        generated_mutation = generate_update_mutation({
+            "mutations": [
+                generate_mutation_data({
+                    "tableName": table,
+                    "returning": returning,
+                    "variables": {
+                        "set": record,
+                        "where": condition,
+                    }
+                }),
+            ],
+            "name": name,
+        })
+
+        m = await self.client.execute_async(generated_mutation['mutation'],
+                                            variable_values=generated_mutation['variables'])
+        data = m.get("m0", [])
+        # del m["m0"]
+        return {**m, "data": data}
+
+    async def delete(self, exp: Union[Dict, int, List[int]], options: Dict = {}) -> Dict:
+        if not exp:
+            raise ValueError('!exp')
+
+        if isinstance(exp, list):
+            where = {"id": {"_in": exp}}
+        elif isinstance(exp, dict):
+            where = self.serialize_where(exp, options.get("table", "links"))
+        else:
+            where = {"id": {"_eq": exp}}
+
+        table = options.get("table", self.table)
+        returning = options.get("returning", self.default_returning(table))
+
+        variables = options.get("variables", {})
+        name = options.get("name", 'delete')
+
+        delete_mutation = generate_delete_mutation({
+            'mutations': [
+                generate_mutation_data({
+                    'tableName': table,
+                    'operation': 'delete',
+                    'returning': returning,
+                    'variables': {
+                        **variables,
+                        'where': where,
+                    }
+                }),
+            ],
+            'name': name,
+        })
+
+        try:
+            m = await self.client.execute_async(delete_mutation['mutation'],
+                                                variable_values=delete_mutation['variables'])
+            data = m.get("object", [])
+            # del m["object"]
+            return {**m, "data": data}
+        except Exception as e:
+            # handle the exception, perhaps re-raise it after logging or cleaning up
+            raise e
+
+    async def serial(self, operations: List[Dict[str, Any]], returning: str = "from_id id to_id type_id value", silent: bool = False):
```

Comment:
Serial should accept object (dict), not arguments directly.

https://github.com/deep-foundation/deeplinks/blob/c6495c15a3abe66392f40121e969859d440cbcef/imports/client.tsx#L805